examples/multipole.py

```diff
@@ -0,0 +1,145 @@
+THICKNESS = 0.1
+IN_RADIUS = 0.3
+OUT_RADIUS = 0.5
+SPACING = 0.05
+
+# boundary configuration
+B_RADIUS = 1.
+B_HEIGHT = 2.
+
+
+# Before v0.8.0
+# 1. Initialize a path
+# 2. Revolve it
+# 3. Add side pieces
+
+multipole_pieces = []
+
+for n in range(1, 2 * ORDER + 1):
+    angle = 2 * np.pi / (2 * ORDER)
+    piece = (
+        G.Path.rectangle_xz(IN_RADIUS, OUT_RADIUS, -THICKNESS/2, THICKNESS/2) 
+        .rotate(Rz=n * angle)
+        .revolve_z(angle)
+        .move(
+            dx=SPACING * np.cos((n + 0.5) * angle),
+            dy=SPACING * np.sin((n + 0.5) * angle),
+        )
+    )
+    piece.name = 'pos' if n % 2 == 0 else 'neg'
+    multipole_pieces.append(piece)
+
+    # We have to add the sides to the pieces of the multipole manually
+    for m in [0, 1]:
+        piece_side = (
+            G.Surface.rectangle_xz(IN_RADIUS, OUT_RADIUS, -THICKNESS/2, THICKNESS/2)
+            .rotate(Rz=(n + m) * angle)
+            .move(
+                dx=SPACING * np.cos((n + 0.5) * angle),
+                dy=SPACING * np.sin((n + 0.5) * angle),
+            )
+        )
+        piece_side.name = piece.name
+        multipole_pieces.append(piece_side)
+
+multipole = G.SurfaceCollection(multipole_pieces)
+
+# To add a boundary we have to:
+# 1. create a path and revolve it
+# 2. manual add a disk to close it at the bottom and top
+boundary = (
+    G.Path.line([1, 0, -1], [1, 0, 1]).revolve_z()
+    + G.Surface.disk_xy(0,0,B_RADIUS).move(dz=-B_HEIGHT/2)
+    + G.Surface.disk_xy(0,0,B_RADIUS).move(dz=B_HEIGHT/2)
+)
+boundary.name = 'boundary'
+
+# Uncomment to compare to new implementation
+# mesh = (multipole + boundary).mesh(mesh_size=0.08)
+# P.plot_mesh(mesh, pos='red', neg='blue', boundary='green')
+# P.show()
+
+#### After v.0.8.0 ###
+# 1. Initialize a surface.
+# 2. Revolve its boundary, cap it off on the other side automatically.
+
+multipole_pieces = []
+
+for n in range(1, 2 * ORDER + 1):
+    angle = 2 * np.pi / ( 2 * ORDER)
+    piece = (
+        G.Surface.rectangle_xz(IN_RADIUS, OUT_RADIUS, -THICKNESS/2, THICKNESS/2) # Initialize a surface
+        .rotate(Rz= n* angle)
+        .revolve_boundary_z(angle, add_cap=True)
+        .move(
+            dx=SPACING * np.cos((n + 0.5) * angle),
+            dy=SPACING * np.sin((n + 0.5) * angle),
+        )
+    )
+    for surface in piece.surfaces:
+        surface.name = 'pos' if n % 2 == 0 else 'neg'
+    
+    multipole_pieces.extend(piece.surfaces)
+
+multipole = G.SurfaceCollection(multipole_pieces)
+
+# creating a boundary can be done in one line with the extrude_boundary function:
+boundary = G.Surface.disk_xy(0, 0, B_RADIUS).move(dz=-B_HEIGHT/2).extrude_boundary([0,0,B_HEIGHT], add_cap=True)
+boundary.name='boundary'
+
+# create mesh
+mesh = (multipole + boundary).mesh(mesh_size=0.07)
+
+# add excitations
+excitation = E.Excitation(mesh, E.Symmetry.THREE_D)
+excitation.add_voltage(pos=1, neg=-1, boundary=0)
+
+# calculate field
+field = S.solve_direct(excitation)
```

Comment:
Consider also plotting the mesh before solving, as a user the program now immediately starts running a heavy simulation without first showing what it's going to compute

Reply:
fixed.